src/apis/impl/AuthenticationApiImpl.ts

```diff
@@ -100,10 +102,69 @@ export class AuthenticationApiImpl implements AuthenticationApi {
     return null
   }
 
-  async initApiAndUserAuthenticationToken(process: AuthenticationProcess, validationCode: string, tokenAndKeyPairProvider: (groupId: string, userId: string) => ([string, [string, string]] | undefined)): Promise<[MedTechApi, ApiInitialisationResult]> {
+  async authenticateAndAskForAccess(userLogin: string, shortLivedToken: string, dataOwnerKeyPair: [string, string] | undefined, tokenAndKeyPairProvider: (groupId: string, userId: string) => ([string, [string, string]] | undefined)): Promise<AuthenticationResult | null> {
+    const authenticationResult = await this.initUserTokenAndCrypto(userLogin, shortLivedToken, dataOwnerKeyPair, tokenAndKeyPairProvider);
+
+    const loggedUser = await authenticationResult.medTechApi.userApi.getLoggedUser();
+    if (!loggedUser) throw new Error("User log in failed");
+    if (!loggedUser.patientId) throw new Error("User is not associated to any Patient");
```

Comment:
I would be more generic and check if the loggedUser has a dataOwner instead of a patient only : That way, if the developer would like to createAndInvite another healthcareParty, he can use both services (createAndInvite & authenticateAndAskAccessToItsExistingData)